### ISSUE TITLE
Easily attach estimated distribution to epidist object

### ISSUE THREAD

**Issue by adamkucharski** [epiverse-trace/epiparameter#310]
In the situation where we have a parameter without a known distribution but a mean & 95% CI (e.g. `epidist_admission_outcome`), we can run `extract_param` to estimate an assumed distribution (e.g. "gamma").

It would be usueful to then be able to easily attach the distribution and estimated parameters to the epidist object, rather than having to use `distributional` to define manually as `epidist_admission_outcome$prob_dist`

This has also been raised by @annecori and @ntncmch, who suggested a neater approach, where the user can simply add a `prob_dist` argument to a modified `as_epidist` function:
```r
as_epidist(serial_interval_faye,prob_dist = "gamma")
```
Then internally, the function calls `convert_summary_stats_to_parameters` and `distributional::dist_gamma()` to attach to the `prob_dist` component of the `epidist` object.

**Comment by adamkucharski** [comment]
✨✨ Here's [an AI-assisted sketch of how you might approach this issue](https://copilot-workspace.githubnext.com/epiverse-trace/epiparameter/issues/310?shareId=dd0685d1-3678-491b-a6bb-70b71d2809d0) saved by @adamkucharski using Copilot Workspace v0.19

<details open><summary><b>Topic</b></summary>
<br/>

Does the `as_epidist` function allow attaching an estimated distribution to the `epidist` object using a `prob_dist` argument?

</details>

<details open><summary><b>Before</b></summary>
<br/>

* No, the `as_epidist` function does not allow attaching an estimated distribution to the `epidist` object using a `prob_dist` argument as per the current implementation in `man/as_epidist.Rd` and `man/as_epidist.data.frame.Rd`. These documentations do not mention a `prob_dist` argument for the function.
* The `as_epidist` function is designed to convert from an R object to an `<epidist>` object, with extra arguments passed to the method not being used, as indicated in `man/as_epidist.Rd`.
* The `as_epidist.data.frame` method specifically converts tabular information in a `<data.frame>` to an `<epidist>`, without reference to attaching a probability distribution using a `prob_dist` argument, as shown in `man/as_epidist.data.frame.Rd`.



</details>
<details open><summary><b>After</b></summary>
<br/>

* Yes, the `as_epidist` function now allows attaching an estimated distribution to the `epidist` object using a `prob_dist` argument.
* The documentation for `as_epidist` in `man/as_epidist.Rd` and `man/as_epidist.data.frame.Rd` is updated to include the `prob_dist` argument.
* The `as_epidist` function internally calls `convert_summary_stats_to_parameters` and the appropriate `distributional::dist_*` function based on the `prob_dist` argument value.
* The `prob_dist` argument supports specifying the distribution type (e.g., "gamma") directly in the `as_epidist` function call.
* The `epidist` object's `$prob_dist` element is now populated with the created distribution object when the `prob_dist` argument is used.


</details>

<details open><summary><b>Plan</b></summary>
<br/>

* `man/as_epidist.Rd` (CHANGE)
  - Add documentation for the `prob_dist` argument, explaining its purpose and usage.
  - Update the `@param` section to include `prob_dist` with a description.
  - Mention in the `@details` section that `prob_dist` can be used to specify the distribution type for parameter estimation.
* `man/as_epidist.data.frame.Rd` (CHANGE)
  - Include `prob_dist` argument in the documentation, specifying that it allows specifying the distribution type directly.
  - Update the `@param` section to document `prob_dist` and its functionality.
  - Highlight in the `@description` or `@details` section how `prob_dist` enhances the function's capability to attach a distribution to the `epidist` object.
* `R/coercion.R` (CHANGE)
  - Implement the `prob_dist` argument in the `as_epidist` and `as_epidist.data.frame` functions.
  - Inside these functions, use `prob_dist` to call `convert_summary_stats_to_parameters` and the appropriate `distributional::dist_*` function.
  - Ensure that the `prob_dist` argument is optional and handled appropriately when not provided.
  - Add error handling for unsupported distribution types specified in `prob_dist`.

</details>

<details open><summary><b>Sketch of implementation</b></summary>
<br/>

[View the changes](https://copilot-workspace.githubnext.com/epiverse-trace/epiparameter/issues/310?shareId=dd0685d1-3678-491b-a6bb-70b71d2809d0)

<details open><summary><b>Details</b></summary>
<br/>

Code analyzed at 7f69ecf377850b59db5ea2a8845ed85f42e2b2ab

</details>

**Comment by joshwlambert** [comment]
Thanks for posting @adamkucharski and for sharing the copilot proposed changes. I've updated to `as_epidist()` and `epireview_to_epidist()` (which is internal and called by `as_epidist()`) in PR #335 to implement the feature requested above. 

Once this feature is merged I'll post a example of the new functionality on the EpiParameter community thread to show how this can be used by the pipeline written by Anne and Anton.

I've also added a new section to the `data_from_epireview.Rmd` vignette in PR #335 to provide users with a demonstration of this new functionality.
